fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+	var/bloodsucker_title						// My Dracula style title
+	var/bloodsucker_reputation					// My "Surname" or description of my deeds
+	// CLAN
+	var/datum/team/bloodsucker/bloodsucker_team
+	var/list/datum/antagonist/vassal/vassals = list()// Vassals under my control. Periodically remove the dead ones.
+	var/datum/mind/creator				// Who made me? For both Vassals AND Bloodsuckers (though Master Vamps won't have one)
+	// POWERS
+	var/list/datum/action/powers = list()// Purchased powers
+	var/poweron_feed = FALSE			// Am I feeding?
+	var/poweron_masquerade = FALSE
+	// STATS
+	var/bloodsucker_level
+	var/bloodsucker_level_unspent = 1
+	var/additional_regen                // How much additional blood regen we gain from bonuses such as high blood.
+	var/bloodsucker_regen_rate = 0.3	// How fast do I regenerate?
+	var/feed_amount = 15				// Amount of blood drawn from a target per tick.
+	var/max_blood_volume = 600			// Maximum blood a Vamp can hold via feeding.
+	// OBJECTIVES
+	var/area/lair
+	var/obj/structure/closet/crate/coffin
+	// TRACKING
+	var/foodInGut 					// How much food to throw up later. You shouldn't have eaten that.
+	var/warn_sun_locker				// So we only get the locker burn message once per day.
+	var/warn_sun_burn 				// So we only get the sun burn message once per day.
+	var/had_toxlover
+	var/level_bloodcost
+	var/passive_blood_drain = -0.1        //The amount of blood we loose each bloodsucker life() tick
+	var/notice_healing                    //Var to see if you are healing for preventing spam of the chat message inform the user of such
+	var/FinalDeath                  //Have we reached final death? Used to prevent spam.
+	// HUD
+	var/valuecolor
+	var/value
+	var/static/list/defaultTraits = list(TRAIT_NOBREATH, TRAIT_SLEEPIMMUNE, TRAIT_NOCRITDAMAGE, TRAIT_RESISTCOLD, TRAIT_RADIMMUNE, TRAIT_NIGHT_VISION, TRAIT_STABLEHEART, TRAIT_NOSOFTCRIT, TRAIT_NOHARDCRIT, TRAIT_AGEUSIA, TRAIT_COLDBLOODED, TRAIT_NOPULSE, TRAIT_VIRUSIMMUNE, TRAIT_HARDLY_WOUNDED, TRAIT_NODISMEMBER)
+
+//This handles the application of antag huds/special abilities
+/datum/antagonist/bloodsucker/apply_innate_effects(mob/living/mob_override)
+	var/mob/living/M = mob_override || owner.current
+	add_antag_hud(antag_hud_type, antag_hud_name, M)
+	M.faction |= ROLE_BLOODSUCKER
+
+//This handles the removal of antag huds/special abilities
+/datum/antagonist/bloodsucker/remove_innate_effects(mob/living/mob_override)
+	var/mob/living/M = mob_override || owner.current
+	remove_antag_hud(antag_hud_type, M)
+	M.faction -= ROLE_BLOODSUCKER
+
+///Called by the add_antag_datum() mind proc after the instanced datum is added to the mind's antag_datums list.
+/datum/antagonist/bloodsucker/on_gain()
+	SSticker.mode.bloodsuckers |= owner // Only add after they've been given objectives
+	SSticker.mode.check_start_sunlight() // Start Sunlight if first Vamp
+	SelectFirstName() // Name & Title
+	SelectTitle(am_fledgling = TRUE) 	// If I have a creator, then set as Fledgling.
```

Comment:
???

this doesn't seem dependent on them having a creator...

Reply:
It's on the old repo, I'll leave it there.